### PR TITLE
Add conditional copy/content and CTA to expanded task items

### LINKED ISSUE #6774: Add task list progression design features to the setup task list.
This is to implement the new designs proposed by @elizaan36 for the setup task list. This will be dependent on completing #6772 and #6773  first.

image

Requirements
* Any task can be expanded at any time (the tasks don't have to be completed in order)
* Only one task can be expanded at a time
* Completed tasks won't expand, they'd go the settings area or task screen where data can be edited

Expanded task copy and CTA:

If store profiler was skipped:
Screen Shot 2021-05-03 at 3 49 59 PM

Add products:
Screen Shot 2021-05-03 at 3 50 12 PM

WCPay installed:
Screen Shot 2021-05-18 at 4 19 30 PM

WCPay not installed:
Screen Shot 2021-05-18 at 4 22 51 PM

Shipping:
Screen Shot 2021-05-03 at 3 51 02 PM

No WCShip, No location set:
Screen Shot 2021-05-18 at 4 24 11 PM

WCShip installed:
Screen Shot 2021-05-03 at 3 50 39 PM

Personalize your store:
Screen Shot 2021-05-18 at 4 19 47 PM

Multiple extensions added:
Screen Shot 2021-05-18 at 4 26 17 PM

One extension added:
Screen Shot 2021-05-18 at 4 26 58 PM

### PR DESCRIPTION
**Pull request by jeffstieler**
Fixes #6774

This PR seeks to implement expanded task item content and CTA buttons, as well as adding the content and CTAs for all of the existing tasks. Some of the tasks have conditional content based on the progress of the store's setup.

A behavioral change was made to the Tax step so that users who can turn on WCS Tax with one click don't have to click "yes please!" twice. You'll see the `auto` query param used for that.

### Accessibility

-   [ ] I've tested using only a keyboard (no mouse)
-   [ ] I've tested using a screen reader
-   [ ] All animations respect [`prefers-reduced-motion`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion)
-   [ ] All text has [at least a 4.5 color contrast with its background](https://webaim.org/resources/contrastchecker/)

### Detailed test instructions:

- Add the `expanded` prop [here](https://github.com/woocommerce/woocommerce-admin/pull/6956#discussion_r641807943) so you see all items as expanded and can verify their copy
- Verify the copy/CTA for all scenarios in https://github.com/woocommerce/woocommerce-admin/issues/6774 - **note that the titles are not changing** even though they differ in the screenshots
- Verify when clicking "yes please" to enable Automated Tax, the Tax step doesn't require any interaction to complete

### CODE REVIEW

File: client/task-list/task-list.js [hunk 1]
```diff
@@ -294,6 +285,9 @@ export const TaskList = ( {
 									onDismiss={ () => dismissTask( task ) }
 									time={ task.time }
 									level={ task.level }
+									action={ task.onClick }
+									actionLabel={ task.action }
+									additionalInfo={ task.additionalInfo }
```
Comment (jeffstieler) [inline client/task-list/task-list.js:290]: Add the `expanded` prop here for testing.

**Review by louwie17 — COMMENTED** [review]
@jeffstieler overall the changes look good. I did run into one issue with the tax task, which I commented on below (easy fix).
I was also curious if the other updates to task list are coming in a separate PR? The handling of expanding/collapsing when clicking on individual task list items, and only going to the task when the action button is clicked (as those items are part of the #6774 requirements).

**Comment by jeffstieler** [comment]
> I was also curious if the other updates to task list are coming in a separate PR? The handling of expanding/collapsing when clicking on individual task list items, and only going to the task when the action button is clicked (as those items are part of the #6774 requirements).

Yes, in https://github.com/woocommerce/woocommerce-admin/pull/7089.

File: client/task-list/tasks/tax.js [hunk 2]
```diff
@@ -46,7 +46,13 @@ class Tax extends Component {
 	}
 
 	componentDidMount() {
+		const { query } = this.props;
+		const { auto } = query;
 		this.reset();
+
+		if ( auto ) {
```
[the remark this answers is not in the document]
Comment (jeffstieler) [reply client/task-list/tasks/tax.js]: Fixed in e996fab0a63e2469b4463ad238c6de54174f3ab8.

**Review by louwie17 — APPROVED** [review]
Thanks for the tax update @jeffstieler, looks great and tested well :) I will have a look at #7089

**Pull request merged**